docs/usage/firmware.md

```diff
@@ -0,0 +1,104 @@
+The vendors of those devices supply a firmware binary and instructions how to flash it.
+
+The firmware updates on a Kubernetes cluster can be done with a special firmware update container that runs on each node.
+This container can either be run manually via SSH or through a Kubernetes _DaemonSet_.
+
+Since the firmware updates are vendor-specific, there are no generic instructions on how to do them.
+To improve this situation and provide a unified and automated experience, the _Linux Vendor Firmware Service_ was started.
+The system service `fwupd` can fetch firmware updates and flash them for supported devices.
+Not all vendors and not all devices are part yet of this effort but it is the only generic mechanism that can be documented as an example.
+
+## Firmware Update Container with fwupd
+
+First we will build a container image that runs fwupd and makes a one time action to update the firmware.
+In the case of fwupd we need a full OS with systemd which makes it a bit more complicated to build and run the container.
+
+First, write the following contents to a `Dockerfile`:
+
+```
+FROM fedora:34
+RUN dnf update -y && dnf -y install fwupd socat && dnf clean all
+# Create a unit file which outputs to the container stdout
+RUN mkdir -p /etc/systemd/system && echo -e '\
+[Unit]\n\
+Description=Firmware Update Action, ends the container after success\n\
+[Service]\n\
+Type=oneshot\n\
+RemainAfterExit=yes\n\
+Restart=on-failure\n\
+RestartSec=5s\n\
+ExecStart=/bin/bash -c "{ fwupdmgr refresh --force && fwupdmgr get-devices && { fwupdmgr get-updates -y || echo Error, no updates ; } && { fwupdmgr update -y || echo Error, failed to update ; } && systemctl poweroff ; } &> /dev/console"\n\
+[Install]\n\
+WantedBy=multi-user.target\n\
+'  > /etc/systemd/system/fwupd-action.service
+RUN systemctl enable fwupd-action.service
+RUN echo OverrideESPMountPoint=/boot >> /etc/fwupd/uefi_capsule.conf
+# Work around a problem with a missing dbus.service unit file
+RUN mkdir -p /etc/systemd/system && ln -fs /usr/lib/systemd/system/dbus-broker.service /etc/systemd/system/dbus.service
+# Mask udev instead of making /sys/ read-only as suggested in https://systemd.io/CONTAINER_INTERFACE/
+RUN systemctl mask systemd-udevd.service systemd-modules-load.service systemd-udevd-control.socket systemd-udevd-kernel.socket
+ENV container=docker
+VOLUME [ "/sys/fs/cgroup" ]
+# Set up a /dev/console link to have the same behavior with or without -it
+CMD [ "/bin/sh", "-c", "if ! [ -e /dev/console ] ; then socat -u pty,link=/dev/console stdout & fi ; exec /sbin/init" ]
+# TODO: maybe need to set up /usr/libexec/fwupd/efi/fwupdx64.efi(.signed) and DisableShimForSecureBoot?
+```
+
+Now build a container image from it and push it to a registry of your choice:
+
+```
+docker build -t my-registry/fwupd:latest .
+docker push my-registry/fwupd:latest
+```
+
+Then, on each node, run it with a Kubernetes _DaemonSet_ as a one-time action or with the following Docker command via SSH:
```

Comment:
```suggestion
Then, on each node, run it with the following Docker command via SSH:
```

I think the phrasing was a bit confusing because it talked about the DaemonSet first, but it referred to the latter example instead.